### ISSUE TITLE
Gap in g:DiffColors list?

### ISSUE THREAD

**Issue by gbcreation** [rickhowe/diffchar.vim#35]
When using these settings in Neovim:

```
vim.api.nvim_set_hl(0, 'DiffCharColor1', { bg = "#FF0000", fg = "#DDDDDD" })
vim.api.nvim_set_hl(0, 'DiffCharColor2', { bg = "#00FF00", fg = "#DDDDDD" })
vim.api.nvim_set_hl(0, 'DiffCharColor3', { bg = "#0000FF", fg = "#DDDDDD" })

vim.g.DiffColors = { 'DiffCharColor1', 'DiffCharColor2', 'DiffCharColor3' }
```

the first color of the list (`DiffCharColor1`) does not seem to be used to highlight the first difference on a line.

![Screenshot_20230519_113914](https://github.com/rickhowe/diffchar.vim/assets/454315/c3cb35f4-8121-49df-ba32-12bccec6e5e0)

On this screenshot, you can see that the text cursor is on the first diff (`g:DiffPairVisible` is set to `3` to show the diff unit in a floating window), but the highlight groups defined in the `g:DiffColors` list begin to be used from the second different word of the lines.

**Comment by rickhowe** [comment]
Currently the 'DiffText' is always used for the 1st diff unit. So your color1, 2, 3 are used for the 2nd, 3rd, 4th ones. I will check and modify in the next update. As a workaround, if you are using version 9.5, change the code in s:GetDiffUnitHL() like this.
from:
`    let hgp += filter(copy(a:dc), '0 < hlID(v:val)')`
to:
`    let hgp = empty(a:dc) ? [s:DCharHL.T] : filter(copy(a:dc), '0 < hlID(v:val)')`

**Comment by rickhowe** [comment]
I have fixed this in version 9.6.
